cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -158,7 +158,7 @@ type NodeConfiguration struct {
 
 // KubeletConfiguration contains elements describing initial remote configuration of kubelet
 type KubeletConfiguration struct {
-	BaseConfig *kubeletconfigv1alpha1.KubeletConfiguration
+	BaseConfig *kubeletconfigv1beta1.KubeletConfiguration
```

Comment:
is kubeadm persisting alpha kubelet configs already? if so, what is its compatibility/upgrade story?

Reply:
I think they feature gated their kubelet config stuff, but I'll ask @luxas

Reply:
kubeadm guards writing Kubelet config with their own gate for dynamic config:
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/init.go#L346
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/init.go#L376
- https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/join.go#L260

and also guards setting defaults for some reason: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1alpha1/defaults.go#L117